vernac/vernacentries.ml

```diff
@@ -2246,6 +2264,14 @@ let interp ?proof ~atts ~st c =
     let using = Option.append using (Proof_using.get_default_proof_using ()) in
     let tacs = if Option.is_empty tac then "tac:no" else "tac:yes" in
     let usings = if Option.is_empty using then "using:no" else "using:yes" in
+    if !Flags.complain_on_proof_without_using_in_section
```

Comment:
This flag is redundant with the warning control system.

Reply:
I was not able to figure out a better way to integrate in the cwarning system. The behavior I want is the following:  when encouterning a "proof" (without "using") inside a section, a warning should only be triggered if the compilation mode is "-vos", which I keep track of using "Flags.complain_on_proof_without_using_in_section". When this mode is used, it should still be possible for the user to turn off such warnings. We could have the warning get triggered for every compilation mode (including .vo production), and ensure that the warning is "Disabled". However, this would cause two issues. First, it is not very clean because it generates many unnecessary calls to the warning system. Second, it is problematic if the user passes the option to turn off the warning in question before passing in the option -vos. In that case, the warning will be Enabled and not Disabled as requested.

Reply:
If you find a better way to proceed, please suggest.

Reply:
>a warning should only be triggered if the compilation mode is "-vos"

Why should -vos change the warnings used?

Reply:
By default, in a -vo compilation, you don't want to report warnings for "Proofs" (without "using") in section. However, by default, in a -vos compilation, you do want to report such warnings.

Reply:
I also don't like this redundancy. Indeed, you don't have the bug related to disabling this warning, but you may have the dual with warnings as errors being ignored depending on the value of this flag.

> a warning should only be triggered if the compilation mode is "-vos", which I keep track of using "Flags.complain_on_proof_without_using_in_section"

That sounds strange. Why not check the compilation mode before emitting the warning?

Reply:
(and not introduce a flag, then)

Reply:
The warning is emitted from vernac.ml. In this file, I have no access to the compilation mode. All the information that I have is that passed to `load_vernac` and via the `Flags` module. So, I could choose to pass the information about the compilation mode in one way or the other, but I think that in the end that still need a boolean reference similar to the one that I use currently. If you confirm that you find it cleaner to pass an argument to `load_vernac`, I can make the change.

Reply:
@maximedenes I manually reverted the spaces to tabs. I did a commit with my changes. The commit hook killed my changes---for good reason, I believe. The hook called it a successful operation, and subsequently deleted the patch file. Hence, my changes are lost. Great.

Reply:
I'm still not convinced that the warning should depend on the compilation mode (and on which IDE you use, for example). Since it is more or less orthogonal to this PR, I'll remove the warning so that we can discuss it while not postponing the integration of the `.vos` file support.